python/cudf/cudf/dataframe/numerical.py

```diff
@@ -65,8 +65,29 @@ def deserialize(cls, header, frames):
         return col
 
     def binary_operator(self, binop, rhs, reflect=False):
+        int_dtypes = [
+            np.dtype("int8"),
+            np.dtype("int16"),
+            np.dtype("int32"),
+            np.dtype("int64"),
+        ]
+        tmp = rhs
+        if reflect:
+            tmp = self
         if isinstance(rhs, NumericalColumn) or np.isscalar(rhs):
             out_dtype = np.result_type(self.dtype, rhs.dtype)
+            if binop in ["mod", "floordiv"]:
+                found_zero_in_column = False
+                if isinstance(tmp, NumericalColumn):
+                    try:
+                        if tmp.find_first_value(0.0):
+                            found_zero_in_column = True
+                    except ValueError:
+                        """This means value not found"""
```

Comment:
Factor this into a separate function and then have this as the last condition in the or statement below. That way we don't do an extra scan of the column unnecessarily.